server/src/main/java/ch/bifrost/server/impl/keyexchange/DHKeyExchangeServer.java

```diff
@@ -0,0 +1,47 @@
+package ch.bifrost.server.impl.keyexchange;
+
+import java.math.BigInteger;
+import java.util.concurrent.TimeUnit;
+
+import com.google.common.base.Optional;
+
+import ch.bifrost.core.api.datagram.DatagramEndpoint;
+import ch.bifrost.core.api.datagram.DatagramMessage;
+import ch.bifrost.core.api.keyexchange.DHKeyExchange;
+import ch.bifrost.core.api.keyexchange.IdKeyPair;
+import ch.bifrost.core.api.keyexchange.Key;
+import ch.bifrost.core.api.keyexchange.DHKeyExchangeRequestMessage;
+import ch.bifrost.core.api.keyexchange.DHKeyExchangeResponseMessage;
+import ch.bifrost.core.api.session.MultiplexingID;
+
+public class DHKeyExchangeServer implements KeyExchangeServer {
+
+	private DatagramEndpoint datagramEndpoint;
+
+	public DHKeyExchangeServer (DatagramEndpoint datagramEndpoint) {
+		this.datagramEndpoint = datagramEndpoint;
+	}
+
+	@Override
+	public Optional<IdKeyPair> get (MultiplexingID id, long timeout, TimeUnit unit) throws Exception {
```

Comment:
I would have hoped :-) that we can make the server key-exchange method independent of whether or not the underlying DatagramEndpoint is multiplexed or not. Basically that by multiplexing we turn one DatagramEndpoint into many DatagramEndpoints, and then we could run the server either on the "basic" DatagramEndpoint or on one of the multiplexed ones. (Of course both client and server must know whether multiplexing is used.)